framework/base/Controller.php

```diff
@@ -238,6 +238,34 @@ public function createAction($id)
         return null;
     }
 
+    /**
+     * Finds action method name by action ID
+     *
+     * @param $id string the action ID
+     * @return string|null action method name
+     * @throws \ReflectionException
+     */
+    private function getMethodNameById($id)
+    {
+        $methodName = 'action' . str_replace(' ', '', ucwords(str_replace('-', ' ', $id)));
+        if (!method_exists($this, $methodName)) {
+            $methods = (new \ReflectionClass($this))->getMethods(\ReflectionMethod::IS_PUBLIC);
+            $methodNames = array_filter(array_map(function ($item) {
```

Comment:
What is happening here?

Reply:
Do you suggest to split it into several lines? :)

Reply:
I would like to know purpose of this code first. Splitting it into separate statements is definitely helpful :) 

I don't really think we should create ambiguous route, and then try to guess method name. We should either leave `_` untouched in route or disallow it completely in action names.

Reply:
I thought of this but we discussed it in the issue comments and decided to leave it this way.